test/index.js

```diff
@@ -1 +1,1159 @@
+          cwd: path.resolve(cwd, 'test'),
+        },
+        expected: {
+          path: path.resolve(cwd, 'README.md'),
+          extension: '.md',
+        },
+      },
+      {
+        name: 'When both 1st and 2nd arguments are empty => null',
+        pathObj: {},
+        defaultObj: {},
+        expected: null,
+      },
+      {
+        name: 'When 1st argument is a string : 1st arg. is treated as ' +
+              '{path: arg}',
+        pathObj: 'test/fixtures/fined',
+        defaultObj: {
+          name: 'app',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When 1st arguments is null => null',
+        pathObj: null,
+        defaultObj: {
+          path: 'test/fixtures/fined',
+          name: 'app',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        expected: null,
+      },
+      {
+        name: 'When 2nd arguments is null : 2nd argument is ignored',
+        pathObj: {
+          path: 'test/fixtures/fined',
+          name: 'app',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        defaultObj: null,
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When both 1st and 2nd arguments is null => null',
+        pathObj: null,
+        defaultObj: null,
+        expected: null,
+      },
+      {
+        name: 'When 1st argument is illegal type => null',
+        pathObj: 123,
+        defaultObj: {
+          path: 'test/fixtures/fined',
+          name: 'app',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        expected: null,
+      },
+      {
+        name: 'When 2nd argument is illegal type : 2nd argument is ignored',
+        pathObj: {
+          path: 'test/fixtures/fined',
+          name: 'app',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        defaultObj: 123,
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When both 1st and 2nd arguments are illegal types => null',
+        pathObj: function() {
+          return {
+            path: 'test/fixtures/fined',
+            name: 'app',
+            extensions: ['.json', '.js'],
+            cwd: cwd,
+            findUp: false,
+          };
+        },
+        defaultObj: true,
+        expected: null,
+      },
+    ],
+  },
+  {
+    name: 'About property \'path\'',
+    cases: [
+      {
+        name: 'When \'path\' is null => null',
+        pathObj: {
+          path: null,
```

Comment:
I think we should always default the things we have, even if path is `null`.  I will try updating this.